indexing-hadoop/src/main/java/io/druid/indexer/InputRowSerde.java

```diff
@@ -105,18 +105,22 @@ public InputRow get()
           }
 
           String t = aggFactory.getTypeName();
-
-          if (t.equals("float")) {
-            out.writeFloat(agg.getFloat());
-          } else if (t.equals("long")) {
-            WritableUtils.writeVLong(out, agg.getLong());
-          } else if (t.equals("double")) {
-            out.writeDouble(agg.getDouble());
+          if (agg.isNull()) {
+            out.writeByte((byte) 1);
```

Comment:
(Optional) you can save some bytes by saving null flag bit array for all aggregations together, near to `WritableUtils.writeVInt(out, aggs.length);` line above

Reply:
Extract 1 as constant